src/index.ts

```diff
@@ -137,25 +144,36 @@ function oAuthPluginServer(
         // Match existing user
         const users = await payload.find({
           collection: slug,
-          where: { [username]: { equals: user[username as 'sub'] } },
+          where: { ['_oauth_username']: { equals: user['sub'] } },
+          showHiddenFields: true,
         })
+
         if (users.docs && users.docs.length) {
           const user = users.docs[0]
           user.collection = slug
           user._strategy = 'oauth2'
+
+          delete user._oauth_username
```

Comment:
the username is in the response already, I only removed the oauth specific one because it is not needed in the following payload.login handler at line 253 anymore